### ISSUE TITLE
Compilation error

### ISSUE THREAD

**Issue by VictorLamoine** [InstitutMaupertuis/davidSDK#10]
Travis builds fine but on other machines I get these strange errors:

```g++
/home/victor/libraries/davidSDK/build/ClientJsonRpc.cpp:33:36: fatal error: davidSDK/ClientJsonRpc.h: No such file or directory
 #include "davidSDK/ClientJsonRpc.h"
                                    ^
compilation terminated.
/home/victor/libraries/davidSDK/src/src/Common.cpp:33:29: fatal error: davidSDK/Common.h: No such file or directory
 #include "davidSDK/Common.h"
                             ^
compilation terminated.
```

```bash
Scanning dependencies of target davidSDK
[  6%] [ 18%] [ 18%] Building CXX object CMakeFiles/davidSDK.dir/json/src/json_value.cpp.o
Building CXX object CMakeFiles/davidSDK.dir/json/src/json_reader.cpp.o
Building CXX object CMakeFiles/davidSDK.dir/json/src/json_writer.cpp.o
[ 25%] Building CXX object CMakeFiles/davidSDK.dir/ClientJsonRpc.cpp.o
[ 37%] [ 37%] Building CXX object CMakeFiles/davidSDK.dir/src/Common.cpp.o
Building CXX object CMakeFiles/davidSDK.dir/src/ErrorCodes.cpp.o
[ 43%] Building CXX object CMakeFiles/davidSDK.dir/src/Exceptions.cpp.o
[ 50%] Building CXX object CMakeFiles/davidSDK.dir/src/IShapeFusion.cpp.o
[ 56%] Building CXX object CMakeFiles/davidSDK.dir/src/IStructuredLightScanner.cpp.o
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp: In constructor ‘david::MotionInfo::MotionInfo()’:
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp:46:4: error: class ‘david::MotionInfo’ does not have any field named ‘angleDeg’
  , angleDeg(0.0)
    ^
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp:47:4: error: class ‘david::MotionInfo’ does not have any field named ‘angleToleranceDeg’
  , angleToleranceDeg(2.0)
    ^
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp: In function ‘david::MotionInfo david::GetMotionInfo_AngleOnly(double, double)’:
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp:74:5: error: ‘struct david::MotionInfo’ has no member named ‘angleDeg’
  mi.angleDeg = angleDeg;
     ^
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp:75:5: error: ‘struct david::MotionInfo’ has no member named ‘angleToleranceDeg’
  mi.angleToleranceDeg = angleToleranceDeg;
     ^
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp: In function ‘david::MotionInfo david::GetMotionInfo_TurntablePartlyCalibrated(double, const double*, double)’:
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp:84:5: error: ‘struct david::MotionInfo’ has no member named ‘angleDeg’
  mi.angleDeg = angleDeg;
     ^
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp:85:5: error: ‘struct david::MotionInfo’ has no member named ‘angleToleranceDeg’
  mi.angleToleranceDeg = angleToleranceDeg;
     ^
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp: At global scope:
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp:95:1: error: ‘SubmeshVertexIndex’ does not name a type
 SubmeshVertexIndex::SubmeshVertexIndex()
 ^
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp:100:1: error: ‘SubmeshVertexIndex’ does not name a type
 SubmeshVertexIndex::SubmeshVertexIndex(int submeshID, int vertexID)
 ^
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp:106:1: error: ‘PosesWithTolerances’ does not name a type
 PosesWithTolerances::PosesWithTolerances()
 ^
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp:115:1: error: ‘CoarseAlignParams’ does not name a type
 CoarseAlignParams::CoarseAlignParams(const MotionInfo& _motionInfo, int _texturePercentage, float _qualityFactor, const ContactRegionPair& _contactRegionPair, const PosesWithTolerances& _blacklist1, const PosesWithTolerances& _whitelist1)
 ^
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp:126:1: error: ‘FineAlignParams’ does not name a type
 FineAlignParams::FineAlignParams(int _maxNumIterations, int _texturePercentage, double _maxTranslation, double _maxRot_deg)
 ^
/home/dell/libraries/davidSDK/src/src/IShapeFusion.cpp:135:1: error: ‘GlobalFineAlignParams’ does not name a type
 GlobalFineAlignParams::GlobalFineAlignParams()
 ^
make[2]: *** [CMakeFiles/davidSDK.dir/src/IShapeFusion.cpp.o] Erreur 1
make[2]: *** Attente des tâches non terminées....
/home/dell/libraries/davidSDK/build/ClientJsonRpc.cpp: In member function ‘david::ServerInfo david::ClientJsonRpc::Connect(const string&, uint16_t)’:
/home/dell/libraries/davidSDK/build/ClientJsonRpc.cpp:83:17: error: expected type-specifier before ‘TcpClient’
  m_client = new TcpClient(ipAddress, port);
                 ^
/home/dell/libraries/davidSDK/build/ClientJsonRpc.cpp:83:17: error: expected ‘;’ before ‘TcpClient’
/home/dell/libraries/davidSDK/build/ClientJsonRpc.cpp:111:19: error: expected type-specifier before ‘TcpClient’
    m_binary = new TcpClient(address, port);
                   ^
/home/dell/libraries/davidSDK/build/ClientJsonRpc.cpp:111:19: error: expected ‘;’ before ‘TcpClient’
make[2]: *** [CMakeFiles/davidSDK.dir/ClientJsonRpc.cpp.o] Erreur 1
/home/dell/libraries/davidSDK/src/src/IStructuredLightScanner.cpp:56:105: error: ‘CalibPoint’ was not declared in this scope
 void IStructuredLightScanner::GetCalibrationError(double& maxDelta, double& rmsError, const std::vector<CalibPoint>& calibPoints)
                                                                                                         ^
/home/dell/libraries/davidSDK/src/src/IStructuredLightScanner.cpp:56:115: error: template argument 1 is invalid
 void IStructuredLightScanner::GetCalibrationError(double& maxDelta, double& rmsError, const std::vector<CalibPoint>& calibPoints)
                                                                                                                   ^
/home/dell/libraries/davidSDK/src/src/IStructuredLightScanner.cpp:56:115: error: template argument 2 is invalid
/home/dell/libraries/davidSDK/src/src/IStructuredLightScanner.cpp:56:129: error: no ‘void david::IStructuredLightScanner::GetCalibrationError(double&, double&, const int&)’ member function declared in class ‘david::IStructuredLightScanner’
 void IStructuredLightScanner::GetCalibrationError(double& maxDelta, double& rmsError, const std::vector<CalibPoint>& calibPoints)
                                                                                                                                 ^
make[2]: *** [CMakeFiles/davidSDK.dir/src/IStructuredLightScanner.cpp.o] Erreur 1
make[1]: *** [CMakeFiles/davidSDK.dir/all] Erreur 2
make: *** [all] Erreur 2
```

**Comment by VictorLamoine** [comment]
https://travis-ci.org/InstitutMaupertuis/davidSDK/jobs/88338172I suspect an old version of CMake going wrong here!

**Comment by VictorLamoine** [comment]
Fixed with #11
